### DISCUSSION TITLE
Listening to Reactive property changes

### DISCUSSION THREAD

**Discussion by ijpatricio** [livewire/livewire#9040] [1. Help]
Hello everyone.

I know I can overcome this issue, but maybe I'm missing an official, more polished way.

`$data` comes from a parent component. It updates well, the `render` updates and the UI as well, all good.

But, I want to have a "listener method", just like `$value` has.

`updatedValue` and `updated` works when `$value` is changed.

When `$data` is changed, only `render` is triggered.

Is this expected behavior? Can't I have an `updatedData` method, and/or have it trigger the `updating/updated` hooks?

```php
<?php

namespace App\Livewire\Lab\RecordsPanel;

use Livewire\Attributes\Computed;
use Livewire\Attributes\Reactive;
use Livewire\Component;

class MeasurementRecordValidation extends Component
{
    #[Reactive]
    public array $data;

    public string $value;

    public function updated($property)
    {
        ray('updated', $property);
    }

    public function updatedData($value, $key)
    {
        ray('updatedData', $value, $key);
    }

    public function updatedValue($value, $key)
    {
        ray('updatedValue', $value, $key);
    }

    public function render()
    {
        ray('render');

        return view('livewire.lab.records-panel.measurement-record-validation');
    }
}

```

```blade
<div>
    {{ json_encode($data) }}


    <button wire:click="$set('value', 'foobar')">Change Value</button>
</div>

```

**Comment by danie-ramdhani** [comment]
workarond: https://github.com/livewire/livewire/discussions/7989#discussioncomment-8551184
